lib/streamlit/testing/element_tree.py

```diff
@@ -163,6 +166,193 @@ def widget_state(self) -> WidgetState:
         return ws
 
 
+@dataclass(init=False)
+class Checkbox(Element, Widget):
+    _value: bool | None
+
+    proto: CheckboxProto
+    type: str
+    id: str
+    label: str
+    help: str
+    form_id: str
+    disabled: bool
+    key: str | None
+
+    root: ElementTree = field(repr=False)
+
+    def __init__(self, proto: CheckboxProto, root: ElementTree):
+        self.proto = proto
+        self.root = root
+        self._value = None
+
+        self.type = "checkbox"
+        self.id = proto.id
+        self.label = proto.label
+        self.help = proto.help
+        self.form_id = proto.form_id
+        self.disabled = proto.disabled
+        self.key = user_key_from_widget_id(self.id)
+
+    def widget_state(self) -> WidgetState:
+        ws = WidgetState()
+        ws.id = self.id
+        ws.bool_value = self.value
+        return ws
+
+    @property
+    def value(self) -> bool:
+        if self._value is not None:
+            return self._value
+        else:
+            state = self.root.session_state
+            assert state
+            return cast(bool, state[self.id])
+
+    def set_value(self, v: bool) -> Checkbox:
```

Comment:
One thing to consider is to wrap set_value with `@property.setter` decorator, so it would be possible
to do things like this `checkbox.value = True` in test code, although I am not sure we need this from an API design perspective

Reply:
I think I don't want that, for 2 reasons. One, `set_value` is intended as a low level primitive that other interaction methods with higher level semantics use as a building block, rather than it being something tests use directly. Two, assignment doesn't work with the fluent api style, due to I think not allowing a return value and not fitting syntactically.